architecture/src/commonMain/kotlin/observable/ReadOnlyPropertyObservable.kt

```diff
@@ -0,0 +1,104 @@
+/*
+ Copyright 2023 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.architecture.observable
+
+import kotlinx.coroutines.flow.Flow
+import kotlin.properties.ReadOnlyProperty
+
+private fun <R : T, T> readOnlyPropertyObservableHelper(readOnlyProperty: ReadOnlyProperty<Any?, T>, observation: Observation<R, T, ObservableOptional.Value<R>>) {
+
+    val readOnlyValue by readOnlyProperty
+
+    observation.beforeObservedValueGet = { current ->
+        @Suppress("UnnecessaryVariable") val new = readOnlyValue // variable actually needed due to delegation
+        if (new != current.valueOrNull)
```

Comment:
We should have linter rule to force brackets for single if/else statements as well

Reply:
There doesnt seem to be a rule for that allows this. Ive manually scanned for them and fixed a bunch